File: backend/schemas/notification.py
from datetime import datetime
from pydantic import BaseModel


class NotificationRead(BaseModel):
    id: int
    uid: str
    component_id: int
    have_read: bool
    release_time: datetime
    type: str


class NotificationReadByUid(BaseModel):
    id: int
    uid: str
    component_id: int
    publisher: str
    course_name: str
    release_time: datetime
    title: str
    content: str
    have_read: bool
    icon_type: str

File: frontend/src/schemas/notification.ts
export interface Notification {
  uid: string,
  release_time: string,
  title: string,
  content: string
}

export interface NotificationCreate {
  have_read: Boolean,
  release_time: string,
  type: string
}

export interface NotificationRead {
  id: number,
  uid: string,
  component_id: number,
  publisher: string,
  course_name: string,
  release_time: string,
  title: string,
  content: string,
  have_read: Boolean,
  icon_type: string,
}  

export interface NotificationUpdate {
  have_read: Boolean
}  
